Tests fail with Go 1.22

Building with 1.21 passes:
```
$ podman run --rm -v "$PWD":/usr/src/gotenv:z -w /usr/src/gotenv golang:1.21 go test ./...
go: downloading golang.org/x/text v0.12.0
go: downloading github.com/stretchr/testify v1.7.5
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading gopkg.in/yaml.v3 v3.0.1
ok  	github.com/subosito/gotenv	0.006s
```
while building with 1.22 fails:
```
$ podman run --rm -v "$PWD":/usr/src/gotenv:z -w /usr/src/gotenv golang:1.22.0 go test ./...
go: downloading golang.org/x/text v0.12.0
go: downloading github.com/stretchr/testify v1.7.5
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/pmezard/go-difflib v1.0.0
--- FAIL: TestScanner (0.00s)
    scanner_test.go:75: 
        	Error Trace:	scanner_test.go:75
        	Error:      	Not equal: 
        	            	expected: 4
        	            	actual  : 3
        	Test:       	TestScanner
        	Messages:   	testCase: regular LF split with trailing LF - expected to have the correct line count
    scanner_test.go:75: 
        	Error Trace:	scanner_test.go:75
        	Error:      	Not equal: 
        	            	expected: 4
        	            	actual  : 3
        	Test:       	TestScanner
        	Messages:   	testCase: regular CR split with trailing CR - expected to have the correct line count
    scanner_test.go:75: 
        	Error Trace:	scanner_test.go:75
        	Error:      	Not equal: 
        	            	expected: 4
        	            	actual  : 3
        	Test:       	TestScanner
        	Messages:   	testCase: regular CRLF split with trailing CRLF - expected to have the correct line count
FAIL
FAIL	github.com/subosito/gotenv	0.006s
FAIL
```

--- a/gotenv.go
+++ b/gotenv.go
@@ -186,7 +186,7 @@ func Write(env Env, filename string) error {
 // If a CR is immediately followed by a LF, it is treated as a CRLF (one single line break).
 func splitLines(data []byte, atEOF bool) (advance int, token []byte, err error) {
 	if atEOF && len(data) == 0 {
-		return 0, nil, bufio.ErrFinalToken
+		return 0, []byte{}, bufio.ErrFinalToken
 	}
 
 	idx := bytes.IndexAny(data, "\r\n")

